lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js

```diff
@@ -55,8 +55,8 @@ const EditCustomFieldsSettings = ({
   permissions,
   history,
 }) => {
-  const makeOkapiRequest = useCallback(makeRequest(okapi)(backendModuleId), [okapi, backendModuleId]);
-  const makeTitleRequest = useCallback(makeRequest(okapi)(), [okapi]);
+  const makeOkapiRequest = useCallback((url) => makeRequest(okapi)(backendModuleId)(url), [okapi, backendModuleId]);
```

Comment:
What is the intent of these changes (and similar ones below)? The before and after look equivalent to me.

Reply:
Functionally, they are equivalent and the change here is just to appease lint. `useCallback()` expects a function and an array, but since we don't use TypeScript, it has no way to recognize that `makeRequest(...)` returns a function (actually, it's a function that returns a function that returns a function that returns a function, which honestly is a bit much if you ask me) and so it whines.

Reply:
Ah yes, the old trick where you have to make your code _less_ clear to persuade Lint that you know what you're doing. I go back and forth on this. I think sometimes it's better just to add an `// eslint-disable-next-line name-of-rule` comment. In this case the change you've made is _probably_ better than that, but it's a tough call.

And yes, a function that returns a function that returns a function that returns a function is just bad design. Functional programming is sold to us on the basis that it's easier to reason about. This kind of behaviour does not help to convey the truth of that.